modules/refined/src/main/scala/medeia/refined/package.scala

```diff
@@ -0,0 +1,16 @@
+package medeia
+
+import cats.syntax.either._
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.refineV
+import medeia.decoder.BsonDecoder
+import medeia.decoder.BsonDecoderError.FieldParseError
+import medeia.encoder.BsonEncoder
+
+package object refined {
+  implicit def encoder[T, P](implicit innerEncoder: BsonEncoder[T]): BsonEncoder[Refined[T, P]] =
+    innerEncoder.contramap(_.value)
+
+  implicit def decoder[T, P](implicit innerDecoder: BsonDecoder[T], validate: Validate[T, P]): BsonDecoder[Refined[T, P]] =
+    bson => innerDecoder.decode(bson).flatMap(t => refineV[P](t).leftMap(error => FieldParseError(error)).toEitherNec)
```

Comment:
Is `FieldParserError` correct?  The refined predicate could also be on e.g. a case class, I think?

Reply:
You are right, thanks. Changed to GenericDecoderError.